src/app/model/Mission.ts

```diff
@@ -208,63 +210,145 @@ class Mission extends BaseModel {
    * Returns all available missions.
    * A mission is available if it has a status of "tentative"
    */
-  getAllAvailable = async () => {
-    const collection = this.getCollection("organizations").doc("1").collection("missions");
-
-    const missionsAvailableForEveryone = await collection
-      .where("status", "==", MissionStatus.tentative)
-      .get();
-
-    if (missionsAvailableForEveryone.docs.length < 1) {
-      return [];
-    }
-    const missions = missionsAvailableForEveryone.docs.map((doc) => doc.data());
-
-    return missions;
-  };
-
-  async create(mission: MissionInterface) {
-    const id = uuidV4();
-
-    console.log(mission);
-    const newMission: MissionInterface = {
-      ...defaultMissionData,
-      ...mission,
-      // TODO change when we figure out dates
-      createdDate: Date.now().toString(),
-      organizationId: Organization.id,
-      id: id,
-    };
-
-    console.log("new", newMission);
-
-    await this.getCollection("organizations")
-      .doc(Organization.id)
+  getAllAvailable = () => {
+    return this.getCollection("organizations")
+      .doc(Organization.uid)
       .collection("missions")
-      .doc(id)
-      .set({
-        ...newMission,
+      .where("status", "==", MissionStatus.tentative)
+      .get()
+      .then((querySnapshot) => {
+        return querySnapshot.docs.map((doc) => doc.data());
       });
-
-    return newMission;
-  }
+  };
 
   /**
    * Update a mision
-   * @param {string} missionId - mission
+   * @param {string} missionUid - mission
    * @param {object} data- updated data
    */
-  update(missionId: string, data: object) {
+  update(missionUid: string, data: object) {
     let sanitized = this.sanitize(data);
     return this.getCollection("organizations")
-      .doc("1")
+      .doc(Organization.uid)
       .collection("missions")
-      .doc(missionId)
+      .doc(missionUid)
       .update({
         ...sanitized,
       });
   }
 
+  /**
+   * create a new mission
+   * returns the new mission id
+   * @param {object} mission
+   * @return {string}
+   */
+  create(mission: MissionInterface) {
```

Comment:
I had to change this method because if we just use `add` it doesn't set the actual `uid` property on the mission object.
Turns out you can get an auto generated id by just generating an empty doc reference and then setting it. (see below)